azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java

```diff
@@ -414,15 +433,76 @@ private static Secret getSecret(final ImageConfig config) {
-            return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withUsername(username).withPasswordSecretRef(passwordName);
+            if (StringUtils.isEmpty(config.registryIdentity)) {
+                final String username = registry.getUserName();
+                final String password = Optional.ofNullable(registry.getPrimaryCredential()).orElseGet(registry::getSecondaryCredential);
+                final String passwordKey = Objects.equals(password, registry.getPrimaryCredential()) ? "password" : "password2";
+                final String passwordName = String.format("%s-%s", registry.getName().toLowerCase(), passwordKey);
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withUsername(username).withPasswordSecretRef(passwordName);
+            } else if (StringUtils.equalsIgnoreCase(config.registryIdentity, "system")) {
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withIdentity("system");
+            } else {
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withIdentity(config.registryIdentity);
+            }
         }
         return null;
     }
 
+    @Nullable
+    private ManagedServiceIdentity getManagedServiceIdentity(ImageConfig imageConfig) {
+        if (StringUtils.isBlank(imageConfig.getRegistryIdentity())) {
+            return null;
+        }
+        if (StringUtils.equalsIgnoreCase(imageConfig.getRegistryIdentity(), "system")) {
+            String principalId = Optional.ofNullable(this.origin)
+                .map(ContainerApp::getIdentity)
+                .filter(identity -> identity.type().equals(ManagedServiceIdentityType.SYSTEM_ASSIGNED) || identity.type().equals(ManagedServiceIdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED))
+                .map(identity -> identity.principalId().toString())
+                .orElseThrow(() -> new AzureToolkitRuntimeException("System managed identity should be enabled before using to pull acr image."));
+            grantACRPullPermissionToIdentity(imageConfig, principalId);
+            return null;
+        }
+        try {
+            Identity identity = Azure.az(AzureManagedIdentity.class).getById(imageConfig.getRegistryIdentity());
+            grantACRPullPermissionToIdentity(imageConfig, identity.getPrincipalId());
+            final String identityJson = String.format("{\"principalId\" : \"%s\", \"clientId\" : \"%s\"}", identity.getPrincipalId(), identity.getClientId());
+            final SerializerAdapter adapter = SerializerFactory.createDefaultManagementSerializerAdapter();
+            // todo: sync with sdk team to find a better way to create user identity object
+            final UserAssignedIdentity userAssignedIdentity = adapter.deserialize(identityJson, UserAssignedIdentity.class, SerializerEncoding.JSON);
```

Comment:
Hmm, not related to this PR, but this is a wired design. It means there is no easy way to create a resource with user assigned managed identity.

The SDK library of appcontainers starting from 1.0.0 has a static method `fromJson`, but that doesn't help either.